packages/fogbender-element/src/FogbenderProvider.tsx

```diff
@@ -1,25 +1,32 @@
+import { createContext, provide } from "component-register";
 import { createNewFogbender, Fogbender } from "fogbender";
-import type { JSX, Accessor, Component } from "solid-js";
-import { createContext, useContext, createSignal, createEffect } from "solid-js";
+import { customElement } from "solid-element";
+import type { JSX } from "solid-js";
 
 export interface FogbenderProviderProps {
-  fogbender?: Accessor<Fogbender>;
-  children?: JSX.Element;
+  fogbender?: Fogbender;
+  children?: JSX.Element[];
 }
 
-export const FogbenderContext = createContext<Accessor<Fogbender>>();
+interface Context {
+  id: symbol;
+  initFn: Function;
+}
+
+const fogbenderContext: Context = createContext((fogbender?: Fogbender) => {
+  return fogbender;
+});
 
-export function getFogbender() {
-  const fogbender = useContext(FogbenderContext);
+customElement<FogbenderProviderProps>(
+  "fogbender-provider",
+  { fogbender: undefined, children: undefined },
+  props => {
+    const defaultFogbender = props.fogbender || createNewFogbender();
 
-  if (!fogbender) {
-    throw new Error("No fogbender set, use FogbenderProvider to set one");
+    provide(fogbenderContext, defaultFogbender);
+
+    return [...(props.children as JSX.Element[])];
```

Comment:
do we need children here?

Reply:
Due to shadow DOM this was not able to render without props.children. Fortunately, I am now able to find a method to not opt for shadow DOM in component-register.
This children code is removed now!